simple_graph/test_wgraph.py

```diff
@@ -0,0 +1,172 @@
+import pytest
+import weighted_graph as wgraph
+from collections import OrderedDict
+
+
+def test___init__():
+    """Test instantiates simple graph with dictionary."""
+    assert wgraph.Wgraph().dict == {}
+
+
+def test_nodes_empty(empty_graph):
+    """Test return list of all nodes in an empty graph"""
+    assert empty_graph.nodes() == []
+
+
+def test_nodes_populated(populated_graph):
+    """Test return list of all nodes in a populated graph"""
+    l = ['a', 'b', 'c', 'd', 'e']
+    sortedlist = populated_graph.nodes()
+    sortedlist.sort()
+    assert sortedlist == l
+
+
+def test_edges(populated_graph):
+    """Test return a list of all edges in the graph"""
+    edges = [('a', 'b'), ('b', 'c'), ('c', 'b'),
+             ('c', 'a'), ('b', 'd'), ('d', 'c')]
+    for edge in edges:
+        assert edge in populated_graph.edges()
+
+
+def test_add_node_empty(empty_graph):
+    """Test adds a new node 'n' to the empty graph"""
+    empty_graph.add_node('a')
+    assert empty_graph.has_node('a')
+
+
+def test_add_node_populated(populated_graph):
+    """Test adds a new node 'n' to the populated graph"""
+    populated_graph.add_node('g')
+    assert populated_graph.has_node('g')
+
+
+def test_has_node(empty_graph, populated_graph):
+    """Test True if node 'n' is contained in the graph, False if not."""
+    assert empty_graph.has_node('a') is False
+    l = ['a', 'b', 'c', 'd', 'e']
+    for node in l:
+        assert populated_graph.has_node(node) is True
+    assert populated_graph.has_node('t') is False
+
+
+def test_add_edge_empty(empty_graph):
+    """Test Adds a new edge to the graph connecting 'n1' and 'n2', if either n1 or n2 are not
+    already present in the graph, they should be added."""
+    empty_graph.add_edge('a', 'b', 7)
+    assert empty_graph.has_node('a')
+    assert empty_graph.has_node('b')
+    assert empty_graph.dict['a'] == {'b': 7}
+
+
+def test_del_node_empty(empty_graph):
+    """Test deletesthenode'n' from the graph, raises an error if no such node exists"""
+    with pytest.raises(AttributeError) as message:
+        empty_graph.del_node('a')
+    assert 'No Such Node Exists' in str(message.value)
+
+
+def test_del_node_populated(populated_graph):
+    """Test deletesthenode'n' from the graph, raises an error if no such node exists"""
+    populated_graph.del_node('a')
+    assert not populated_graph.has_node('a')
+    assert 'a' not in populated_graph.dict['c']
+
+
+def test_del_edge_empty(empty_graph):
+    """Test deletes the edge connecting non-existing 'n1' and 'n2' from the graph,
+    raises an error if no such edge exists"""
+    with pytest.raises(AttributeError) as message:
+        empty_graph.del_edge('a', 'b')
+    assert 'No Such Edge Exists' in str(message.value)
```

Comment:
might be a bit confusing binding the `Exception` raised by deleting a non-existent edge to the symbol `message`.  The error *has* a message, but is not itself a message, no?